src/main/java/io/cryostat/net/web/http/api/beta/ProbeTemplateGetHandler.java

```diff
@@ -0,0 +1,157 @@
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
+import io.cryostat.net.web.http.api.v2.RequestParameters;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpMethod;
+
+public class ProbeTemplateGetHandler extends AbstractV2RequestHandler<String> {
+
+    static final String PATH = "probes";
+
+    private final Logger logger;
+    private final NotificationFactory notificationFactory;
+    private final LocalProbeTemplateService probeTemplateService;
+    private final FileSystem fs;
+    private final Gson gson;
+    private static final String NOTIFICATION_CATEGORY = "GetProbeTemplates";
+
+    @Inject
+    ProbeTemplateGetHandler(
+            AuthManager auth,
+            NotificationFactory notificationFactory,
+            LocalProbeTemplateService probeTemplateService,
+            Logger logger,
+            FileSystem fs,
+            Gson gson) {
+        super(auth, gson);
+        this.notificationFactory = notificationFactory;
+        this.logger = logger;
+        this.probeTemplateService = probeTemplateService;
+        this.fs = fs;
+        this.gson = gson;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public IntermediateResponse<String> handle(RequestParameters params) throws Exception {
+        List<String> probeTemplates = new ArrayList<>();
+        try {
+            for (ProbeTemplate p : this.probeTemplateService.getTemplates()) {
+                probeTemplates.add(
+                        "{\"name\" : \""
+                                + p.getFileName()
+                                + "\" , "
+                                + "\"xml\" : \""
+                                + p.serialize().replaceAll("\n", "").replaceAll("\"", "")
+                                + "\"}");
+            }
+            notificationFactory
+                    .createBuilder()
+                    .metaCategory(NOTIFICATION_CATEGORY)
+                    .metaType(HttpMimeType.JSON)
+                    .message("Probe templates fetched")
+                    .message(probeTemplates.toString())
+                    .build()
+                    .send();
+        } catch (Exception e) {
+            throw new ApiException(400, e.getMessage(), e);
+        }
+        return new IntermediateResponse<String>().body(probeTemplates.toString());
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.DELETE_PROBE_TEMPLATE);
```

Comment:
I think this should be `ResourceAction.READ_PROBE_TEMPLATE`, which looks like it needs to be added to the enum.

Reply:
It should, good catch, I've updated this now.